oop1-construct.js

```diff
@@ -2,105 +2,323 @@
+
+    calcPerimeter() {
+        if (this.name === 'triangle') {
+            console.log(`The perimeter of a ${this.name} is ${(this.base) + (this.sides[1]) + this.sides[2]}`)
+        }
+        else if (this.name === 'rectangle') {
+            console.log(`The perimeter of a ${this.name} is ${(this.length * 2) + (this.width * 2)}`)
+        }
+    }
+
+    calcCircumference() {
+        if (this.name === 'circle') {
+            console.log(`The circumference of a ${this.name} is ${(Math.PI * this.radius * 2).toFixed(2)}`)
+        }
     }
 }
 
-triangle.base = triangle.sides[0];  //base is the 1st side in the triangle.sides array.
+
+const newTriangle = new Shape("triangle", [4,7,7], null, null, null, null, null);
+newTriangle.base = newTriangle.sides[0];
+newTriangle.height = newTriangle.calcHeight();
+
+console.log(newTriangle)
+newTriangle.calcArea()
+newTriangle.calcPerimeter()
+
+const newRectangle = new Shape ('rectangle', 4, null, null, 2, 5, null);
+
+console.log(newRectangle)
+newRectangle.calcArea()
+newRectangle.calcPerimeter()
+
+const newCircle = new Shape('circle', 1, null, null, null, null, 5);
+
+console.log(newCircle)
+newCircle.calcArea()
+newCircle.calcCircumference()
+
+
+
+const triangle = {
+  name: 'triangle',
+  sides: [4, 7, 7], //lengths of each side
+  base: null,
+  height: null, //determine this height using basic geometry.  You may need to research this one.
+  calcArea: function () {
+    console.log(`${this.name}'s area is calculated to be : ${0.5 * this.base * this.height}`);
+  },
+  calcPerimeter: function () {
+    console.log(`${this.name}'s perimeter is calculated to be : ${this.base + (this.sides[1] + this.sides[2]) }`);
+  },
+};
+
+
+triangle.base = triangle.sides[0]; //base is the 1st side in the triangle.sides array.
 triangle.height = 6.71; //determine the height using basic geometry.  How do you calculate the height of a triangle with 2 equal sides?  If you use the formula to caluclate the height, this is a bonus.  If you hard code the correct value here, that will be sufficient, but no bonus :(
 
 console.log(triangle);
 triangle.calcArea();
 triangle.calcPerimeter();
 
+
 const rectangle = {
-    name : "rectangle",
-    sides : 4,
-    length : 2,
-    width : 5,
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${this.length * this.width}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(2 * this.length) + (2 * this.width) }`)
-    }
-}
+  name: 'rectangle',
+  sides: 4,
+  length: 2,
+  width: 5,
+  calcArea: function () {
+    console.log(`${this.name}'s area is calculated to be : ${this.length * this.width}`);
+  },
+  calcPerimeter: function () {
+    console.log(`${this.name}'s perimeter is calculated to be : ${2 * this.length + 2 * this.width}`);
+  },
+};
 
 console.log(rectangle);
 rectangle.calcArea();
 rectangle.calcPerimeter();
 
+
+const newCircle = new Shape("circle")
```

Comment:
This line causes an error to be thrown because newCircle has already been declared on line 124